src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/QueryStringSegment.cs

```diff
@@ -7,7 +7,14 @@ public class QueryStringSegment : PatternSegment
     {
         public override string Evaluate(RewriteContext context, MatchResults ruleMatch, MatchResults condMatch)
         {
-            return context.HttpContext.Request.QueryString.ToString();
+            var queryString = context.HttpContext.Request.QueryString.ToString();
```

Comment:
Just to be sure queryString can never be null, right? If so you can probably use IsEmpty instead/